src/moa/moa_controllers/moa_controllers/ack_to_can.py

```diff
@@ -0,0 +1,111 @@
+#/usr/bin/env python3
+from typing import Optional
+import rclpy
+from rclpy.node import Node
+import time
+
+from ackermann_msgs.msg import AckermannDriveStamped
+from ackermann_msgs.msg import AckermannDrive
+from moa_msgs.msg import CANStamped
+from std_msgs.msg import Header
+from builtin_interfaces.msg import Time
+
+import numpy as np
+import zlib
+from math import pi
+
+
+def compress_floats(values:np.ndarray) -> bytes:    
+    # Compute the differences between consecutive values
+    differences = np.diff(values)
+
+    # Compress the differences using zlib
+    compressed = zlib.compress(differences, level=9)
+
+    return compressed
```

Comment:
How does this compression function actually work bc you will have to code the decompression algorithm on the motec by hand in c

Reply:
its from a library (https://github.com/madler/zlib) that seems to have already got C implementation

Reply:
so i think it would transfer over

Reply:
Im fine with that but I also think that it should be easy to look at the can message and see what data is being transfered, as well this function should be as fast as posible, but if you think that this is justified then we can go through with using this

Reply:
yea thats a good point. When testing the parser function just to check size (by using sys.getsizeof and len() of bytes object output) I kept getting a minimum size of 12 bytes after compression. That's really the only reason I used the compression anyway.